### PR TITLE
LibGL: Fix texture sampling texel coordinate calculation

### PR DESCRIPTION
**Pull request by sunverwerth**
Previously we multiplied the interpolated texture coordinates by
width - 1 and height - 1 instead of width and height which resulted in
some wrongly mapped textures, especially visible in the glquake light
maps.

This also corrects the wrap mode being wrongly swapped for s/t
coordinates.

The wrap mode calculation for mirrored wrap and clamp needed to be
adjusted to always return values < 1.0f to make sure we are not sampling
out of bounds.

Before:
Bildschirmfoto zu 2021-12-12 23-34-40

After:
Bildschirmfoto zu 2021-12-12 23-32-50

### CODE REVIEW

File: Userland/Libraries/LibGL/Tex/Sampler2D.cpp [hunk 1]
```diff
@@ -21,12 +21,12 @@ static constexpr float wrap_mirrored_repeat(float value)
...
     float frac = value - integer;
     bool iseven = fmodf(integer, 2.0f) == 0.0f;
-    return iseven ? frac : 1 - frac;
+    return iseven ? frac : 1 - frac - NumericLimits<float>::epsilon();
 }
 
 static constexpr float wrap_clamp(float value)
 {
-    return clamp(value, 0.0f, 1.0f);
+    return clamp(value, 0.0f, 1.0f - NumericLimits<float>::epsilon());
```
Comment (GMTA) [inline Userland/Libraries/LibGL/Tex/Sampler2D.cpp:29]: On page 272 of [the spec](https://www.khronos.org/registry/OpenGL/specs/gl/glspec15.pdf) it is mentioned that the range is exactly `[0, 1]` and the texture's border should be taken into account when sampling - I don't think that is implemented yet.

File: Userland/Libraries/LibGL/Tex/Sampler2D.cpp [hunk 2]
```diff
@@ -21,12 +21,12 @@ static constexpr float wrap_mirrored_repeat(float value)
     float integer = floorf(value);
     float frac = value - integer;
     bool iseven = fmodf(integer, 2.0f) == 0.0f;
-    return iseven ? frac : 1 - frac;
+    return iseven ? frac : 1 - frac - NumericLimits<float>::epsilon();
```
Comment (GMTA) [inline Userland/Libraries/LibGL/Tex/Sampler2D.cpp:24]: See my other comment on clamping

File: Userland/Libraries/LibGL/Tex/Sampler2D.cpp (hunk 1 above)
Comment (GMTA) [reply Userland/Libraries/LibGL/Tex/Sampler2D.cpp:29]: Maybe you could create a special case for `1.0` texture coordinates after wrapping and add a `FIXME`?

Comment (sunverwerth) [reply Userland/Libraries/LibGL/Tex/Sampler2D.cpp]: We use `GL_CLAMP_TO_EDGE` behaviour for now.

File: Userland/Libraries/LibGL/Tex/Sampler2D.cpp (hunk 2 above)
Comment (sunverwerth) [reply Userland/Libraries/LibGL/Tex/Sampler2D.cpp]: We use `GL_CLAMP_TO_EDGE` behaviour for now.

**Comment by GMTA** [comment]
LGTM!

**Pull request merged**